bootstrap/bootstrap.go

```diff
@@ -209,47 +209,57 @@ func (b *Bootstrap) Cancel() error {
 	return nil
 }
 
+type HookConfig struct {
+	Name           string
+	Scope          string
+	Path           string
+	Env            env.Environment
+	SpanAttributes map[string]string
+}
+
 // executeHook runs a hook script with the hookRunner
-func (b *Bootstrap) executeHook(ctx context.Context, scope string, name string, hookPath string, extraEnviron env.Environment) error {
-	span, ctx := tracetools.StartSpanFromContext(ctx, "hook.execute", b.Config.TracingBackend)
+func (b *Bootstrap) executeHook(ctx context.Context, hookCfg HookConfig) error {
+	spanName := b.implementationSpecificSpanName(fmt.Sprintf("hook.%s.%s", hookCfg.Scope, hookCfg.Name), "hook.execute")
```

Comment:
Oh, I think we might need to map legacy/internal `hookCfg.Scope` names to [publicly documented hook location names](https://buildkite.com/docs/agent/v3/hooks#hook-locations).

- `local` → `repository`
- `global` → `agent`

Reply:
(and we're going to change the format to e.g. `agent environment hook`, right?)

Reply:
yep yep! all done now